artif: dump /etc/ld.so.preload with debugfs/xfs_db

mnrkbys opened the pull request:
Add a new artifact to collect /etc/ld.so.preload.
Although LD_PRELOAD rootkits may hide /etc/ld.so.preload, it can be read via debugfs.

mnrkbys committed:
artif: /etc/ld.so.preload via debugfs

Add a new artifact to collect /etc/ld.so.preload.
Although LD_PRELOAD rootkits may hide /etc/ld.so.preload, it can be read via debugfs.

mnrkbys committed:
artif: Dump /etc/ld.so.preload with xfs_db

Add a new artifact to dump /etc/ld.so.preload.
If the file system where /etc is located is XFS, we need to use xfs_db instead of debugfs.

mnrkbys committed:
artif: modify command

mnrkbys committed:
refactor: tidy up linux_dump_ldsopreload.sh

Refactor command options.

mnrkbys committed:
refactor: suppress the status message of dd

mnrkbys committed:
artif: merge artifacts

Merge debugfs.yaml and xfs_db.yaml into one file.

mnrkbys committed:
refactor: optimize awk code

mnrkbys committed:
refactor: calculate offset based on blocksize

mnrkbys committed:
refactor: improve versatility

Any file can be dumped.

mnrkbys committed:
refactor: fix problems detected by ShellCheck

Fix problems detected by ShellCheck and a bug regarding tracking of symbolic links

mnrkbys committed:
refactor: remove the mount point string from file path on EXT4 filesystem

tclahr commented:
This is cool, but I need some time for testing it. I will release UAC v3.0.0 and move this to 3.1.0. Thanks!

tclahr committed:
Merge branch 'develop' into ld_so_preload

tclahr committed:
Merge branch 'develop' into ld_so_preload

tclahr committed:
Merge branch 'develop' into ld_so_preload

tclahr committed:
Merge branch 'develop' into ld_so_preload

tclahr committed:
artif: add stat and rename files

Add collection of stat even if the file is hidden.
Change the output file as any files that are written to disk that matches ld.so.preload is modified by the rootkit.
Moved the condition from command to condition.

tclahr commented:
Before merging into develop, I had to do some small changes.

1. During some tests, I noticed that any files that contain the string `ld.so.preload` is hooked by the rootkit, so UAC (or any other tool) cannot write to files like etc_ld.so.preload.txt. So I changed the output_file to etc_ld_so_preload.txt

2. I changed your .sh script name to reflect what it is doing.

3. I moved the condition to check the file system from the `command` to `condition` property. This is due to the fact even systems that use xfs also have debugfs installed as /boot uses to be formatted as ext. So I think it would be better checking the file system among with the debugfs in the condition property.

4. I noticed that stat can retrieve file stats for the /etc/ld.so.preload even if the rootkit is enabled. So I added a collector for that.

5. I added %mount_point% in the command as this artifact can also be run in offline disks.

tclahr committed:
Update hidden_etc_ld_so_preload.yaml

Change script name to linux_dump_etc_ld_so_preload.sh

mnrkbys commented:
Thanks for your detailed review.

tclahr committed:
Merge branch 'develop' into ld_so_preload
